### DISCUSSION TITLE
#define LATITUDE / #define LONGITUDE redefine (and others) in user_config_override.h does not work?

### DISCUSSION THREAD

**Discussion by forumlw06** [arendst/Tasmota#20803] [Q&A] [answered]
Hello,

i am trying to build a special version for my Shelly 2.5 with some predefined values in Tasmota. I am using Tasmocompiler and add the following as additional parameters (will be added to user_config_override.h):
```
#ifdef CODE_IMAGE_STR
	#undef CODE_IMAGE_STR
#endif
#define CODE_IMAGE_STR "Shelly25_latlon"

#define LATITUDE               48.548981049999995
#define LONGITUDE              9.817778139882574
```

It compiles well and if i flash the binary i see following output in console (i deleted the uninteresting parts)
```
00:00:00.001 HDW: ESP8266EX
00:00:00.015 UFS: FlashFS mounted with 968 kB free
00:00:00.078 CFG: Loaded from File, Anzahl 889
00:00:00.133 QPC: Reset
00:00:00.192 Projekt tasmota - Rolladen Test Version 13.4.0(Shelly25_latlon)-2_7_6(2024.02.24 22:37:26)

23:40:35.042 CMD: longitude
23:40:35.047 RSL: RESULT = {"Longitude":2.294442}

```

**So looks like my CODE_IMAGE_STR was used, but my LAT/LON definitions sem to be the default.**

Same is true if i try to increase the resolution for temperature/humidity (#define TEMP_RESOLUTION        3) (#define HUMIDITY_RESOLUTION    3). If i check the Tempres/humres values in console they both are 1 (default) and not 3

Any idea, where my mistake is?

Thanks!

Regards Rainer

**Comment by sfromis** [comment, marked as answer]
Changed *defaults* for values you can config by commands will not be used when upgrading a binary, unless you do a full reset.

**Comment by barbudor** [comment]
Those are initialization values that Tasmota will use when flashed the 1st time or in case of a full config reset (command 'reset 1')

If you upgrade to that firmware while you have already ran tasmota on the device, it will use the default values that you can still change through commands

**Reply by forumlw06** [reply]
Thank you for the explanation. On second thought that makes perfect sense. I was just puzzled and wondering if i am doing some thing wrong.
Pretty new to all that interesting stuff
